addons/website_sale/controllers/reorder.py

```diff
@@ -4,78 +4,68 @@
 from odoo.http import request, route
 
 from odoo.addons.sale.controllers import portal as sale_portal
+from odoo.addons.website_sale.controllers.cart import Cart
 
 
 class CustomerPortal(sale_portal.CustomerPortal):
 
     def _sale_reorder_get_line_context(self):
         return {}
 
-    def _get_common_order_line_data(self, line, add_to_cart_allowed=True):
-        combination = (
-            line.product_id.product_template_attribute_value_ids
-            | line.product_no_variant_attribute_value_ids
-        )
-        return {
-            'product_template_id': line.product_id.product_tmpl_id.id,
-            'product_id': line.product_id.id,
-            'combination': combination.ids,
-            'no_variant_attribute_value_ids': line.product_no_variant_attribute_value_ids.ids,
-            'product_custom_attribute_values': [
-                {
-                    'custom_product_template_attribute_value_id': pcav.custom_product_template_attribute_value_id.id,
-                    'custom_value': pcav.custom_value,
-                }
-                for pcav in line.product_custom_attribute_value_ids
-            ],
-            'quantity': line.product_uom_qty,
-            'combinationInfo': line.product_id.product_tmpl_id.with_context(
-                **self._sale_reorder_get_line_context()
-            )._get_combination_info(combination, line.product_id.id, line.product_uom_qty)
-            if add_to_cart_allowed else {},
-        }
-
     @route(
-        '/my/orders/reorder_modal_content',
+        '/my/orders/reorder',
         type='jsonrpc',
         auth='public',
         website=True,
-        readonly=True,
     )
-    def my_orders_reorder_modal_content(self, order_id, access_token):
+    def my_orders_reorder(self, order_id, access_token):
+        """ Retrieve reorder content and automatically add products to the cart.
+
+        param int order_id: The ID of the sale order to reorder.
+        param str access_token: The access token for the sale order.
+        return: The number of items in the cart after reordering.
```

Comment:
nitpicking: No need to be too specific here :)
```suggestion
        return: The number of items in the cart.
```